ocaml/typing/ctype.ml

```diff
@@ -2032,82 +2032,103 @@ let rec estimate_type_jkind env ty =
   | Tpoly (ty, _) -> estimate_type_jkind env ty
   | Tpackage _ -> Jkind (value ~why:First_class_module)
 
-
-(* The ~fixed argument controls what effects this may have on `ty`.  If false,
-   then we will update the jkind of type variables to make the check true, if
-   possible.  If true, we won't (but will still instantiate sort variables).
-
-   The "fuel" argument here is used because we're duplicating the loop of
+(**** checking jkind relationships ****)
+
+type type_jkind_sub_result =
+  | Success
+    (* The [Type_var case might still be "success"; caller should check.
```

Comment:
missing `]` here.

Also would it be better to factor this type to `(unit, type_jkind_sub_error) Result.t`?

Reply:
Thanks about the `]`.

For the refactor: I don't think so, mostly because `Type_var` is neither success nor failure. The caller needs to figure out how to treat that.

Reply:
That makes sense.